test/e2e/framework/fixture.go

```diff
@@ -71,13 +71,17 @@ func (f *KCPFixture) SetUp(t *testing.T) func() {
 
 		// Wait for the server to become ready
 		go func(s *kcpServer) {
+			defer wg.Done()
 			err := s.Ready()
 			require.NoErrorf(t, err, "kcp server %s never became ready: %v", s.name, err)
-			wg.Done()
 		}(srv)
 	}
 	wg.Wait()
 
+	if t.Failed() {
+		t.Fatal("Fixture setup failed: one or more servers did not become ready")
```

Comment:
do we see the error?

Reply:
Yes. The `require` call will log an error and mark the `testing.T` as failed. But since it's called in a goroutine it's panic won't propagate. Checking `t.Failed` here ensures that the parent can die in response to the assertion failure in the goroutine.

Reply:
ack